src/services/mentor.service.ts

```diff
@@ -0,0 +1,75 @@
+import { dataSource } from '../configs/dbConfig'
+import Mentor from '../entities/mentor.entity'
+import Profile from '../entities/profile.entity'
+import { ApplicationStatus } from '../enums'
+import Category from '../entities/category.entity'
+
+export const createMentor = async (req: any) => {
+  const userId = req.user.uuid
+  const userApplication = req.body
+
+  try {
+    const profileRepository = dataSource.getRepository(Profile)
+    const userProfile = await profileRepository.findOne({
+      where: { uuid: userId }
+    })
+
+    if (!userProfile) {
+      throw new Error('User profile not found')
+    }
+
+    const mentorRepository = dataSource.getRepository(Mentor)
+    const existingMentor = await mentorRepository.findOne({
+      where: { uuid: userId }
+    })
+
+    if (existingMentor) {
+      if (existingMentor.state === ApplicationStatus.PENDING) {
+        return 'Application pending'
+      } else {
+        const categoryRepository = dataSource.getRepository(Category)
+        const category = await categoryRepository.findOne({
+          where: { uuid: '60b5b847-99a2-4e47-b35b-81b4284311dd' }
+        })
+
+        if (!category) {
+          throw new Error('User profile not found')
+        }
```

Comment:
category id will be a part of the mentor application. 

` where: { uuid:  application.category}`

Reply:
In the API doc `application` has 3 questions only. does it also have category id?

Reply:
It is just an example. But it will have the category id for sure.